### PR TITLE
update with labels instead of hashes

### PR DESCRIPTION
**Pull request by meik99**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -79,3 +79,4 @@ tags
 /debug.sh
 /tmp/
 /reapply-all.sh
+
```
Comment (DTMad) [inline .gitignore:82]: Newline?

**Review by DTMad — DISMISSED** [review]

File: .gitignore (hunk 1 above)
Comment (meik99) [reply .gitignore:82]: lines should end in newlines / linebreaks according to [POSIX convention](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap03.html#tag_03_206)

File: pkg/controller/activegate/version_label.go [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+			imagePullSecret, err := dao.GetImagePullSecret(r.client, pod)
+			if err != nil {
+				// Something wrong with pull secret, exit function entirely
+				return err
+			}
+
+			dockerConfig, err := parser.NewDockerConfig(imagePullSecret)
+			// If an error is returned, try getting the image anyway
+
+			versionLabel, err2 := version.GetVersionLabel(status.Image, dockerConfig)
```
Comment (lrgar) [inline pkg/controller/activegate/version_label.go:30]: Do you actually get something on dockerConfig is there was an error? Or just using nil there?

File: pkg/controller/activegate/activegate_controller.go [hunk 3]
```diff
@@ -142,12 +142,25 @@ func (r *ReconcileActiveGate) Reconcile(request reconcile.Request) (reconcile.Re
...
+	}
+
+	pods, err := r.findPods(instance)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	err = r.setVersionLabel(pods)
+	if err != nil {
+		if _, isStatusError := err.(*errors.StatusError); isStatusError {
```
Comment (lrgar) [inline pkg/controller/activegate/activegate_controller.go:157]: For reference there is `errors.As()` on the Go library. Although probably better to do this check before creating the StateFul set?

File: pkg/controller/activegate/version_label.go (hunk 2 above)
Comment (meik99) [reply pkg/controller/activegate/version_label.go:30]: On error, `dockerConfig` is `nil`

**Review by lrgar — APPROVED** [review]

**Pull request merged**